manifest/fusing_provider.go

```diff
@@ -60,41 +58,92 @@ func (m *FusingManifestProvider) ManifestUpdates() <-chan *Manifest {
 }
 
 func (m *FusingManifestProvider) Start(ctx context.Context) error {
-	head, err := m.ec.GetHead(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to determine current head epoch")
+	if err := m.priority.Start(ctx); err != nil {
+		return err
 	}
 
-	switchEpoch := m.static.BootstrapEpoch - m.static.EC.Finality
-	headEpoch := head.Epoch()
-
-	if headEpoch >= switchEpoch {
-		m.manifestCh <- m.static
-		return nil
+	priorityManifest := <-m.priority.ManifestUpdates()
```

Comment:
I would change this such that the switch between secondary won't happen until at least one non-nil manifest is recieved from primary. This way we  will remove the special case of `nil` manifest from primary.